articles/group-operator/group.md

```diff
@@ -0,0 +1,154 @@
+
+Let’s take a look at its syntax:
+
+`{ $group: { _id: <expression>, <field1>: { <accumulator1> : <expression1> }, ... } }`
+
+The `_id` field in the output documents contains the distinct group by key. The output documents can also contain computed fields that hold the values of some accumulator expression grouped by the $group‘s `_id` field.
+
+Basically, the `_id` field here will take those fields by which you want to group the documents. It is mandatory to use `_id` here.
+
+The $group operator does not remove the data from the database. It just aggregates multiple documents into a single document based on the field passed to the `_id` field.
+
+### Important points to remember
+
+- It can't be used with the `LogReduce` operator.
+
+- When parsing and naming (aliasing) fields, avoid using the names of grouping functions or other operators as field names.
+
+- When using count, or any grouping function, remember to include the underscore before the field name (sort by _count).
+
+- Multiple aggregation functions can be on the same line but you can't include another function, such as a math function, on the same line of a query.
+### Example
+
+Now, let’s look at an example.
+
+Suppose we have a collection of 3 people with their names, gender, age and hobbies:
+
+Let's create a database:
+```bash 
+use review 
+```
+Now, let's create a collection named `person` and insert multiple documents in that collection using `insertMany()` function like:
+```bash
+db.person.insertMany([
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbd"),
+        "name" : "Harit Joshi",
+        "age" : 18,
+        "gender" : "Male",
+        "hobbies" : [
+                "sports",
+                "cooking",
+                "gaming"
+        ]
+    },
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbe"),
+        "name" : "Maria Swartz",
+        "age" : 27,
+        "gender" : "Female",
+        "hobbies" : [
+                "sports",
+                "swimming",
+                "gaming"
+        ]
+    },
+    {
+        "_id" : ObjectId("5ffb0dd58591ec5a52d2afbf"),
+        "name" : "Billy Marton",
+        "age" : 34,
+        "gender" : "Male",
+        "hobbies" : [
+                "singing",
+                "cooking",
+                "jogging"
+        ]
+    }
+])
+```
+Now that we have some documents in the collection, we can use the $group to find how many males and how many females are there. In this short example, it is pretty clear that there are 2 males and 1 female.
+
+```bash
+db.person.aggregate([ { $group: {_id: {gender: "$gender"}, totalPeople: {$sum: 1}} } ]).pretty()
+```
+
+Output:
+
+```bash
+{ "_id" : { "gender" : "Female" }, "totalPeople" : 1 }
+{ "_id" : { "gender" : "Male" }, "totalPeople" : 2 }
+```
+
+For the `$group` operator to work, you should use an accumulator operator like `$sum`, `$avg`, `$max`, etc since the operator yields a combined result based on the grouping expression.
```

Comment:
@HARITJOSHI1 Please add a link to another article or documentation that explains other accumulator operators.